### DISCUSSION TITLE
How to implement multi-tenant with IdentityServer

### DISCUSSION THREAD

**Discussion by shivakb** [kgrzybek/modular-monolith-with-ddd#312] [General]
I built a REST API based on your work, and we are building additional functionality on top of it. Now we need to provide the API as SAAS, which means multi-tenant support. How should we go about implementing?

**Comment by sbalbinot** [comment]
If you're using shared database approach you could just pass a tenantId through http context and use efcore filters to filter for it. But with a database per tenant approach I don't know, because each tenant would need to have background workers for processing inbox, outbox and internal commands. Depending on the number of tenants this could scale up quickly. Actually, if you managed to implement this, I'm curious on how you did it.
